### PR TITLE
Implement vfs with AlibabaCloud OSS

### PR DESCRIPTION
**Pull request by xh4n3**
Add new vfs implementation to kops with Aliyun(AlibabaCloud) OSS.
This is a part of https://github.com/kubernetes/kops/issues/4127

### CODE REVIEW

File: util/pkg/vfs/context.go [hunk 1]
```diff
@@ -348,3 +355,27 @@ func (c *VFSContext) buildOpenstackSwiftPath(p string) (*SwiftPath, error) {
...
+	bucket := strings.TrimSuffix(u.Host, "/")
+	if bucket == "" {
+		return nil, fmt.Errorf("invalid aliyun oss path: %q", p)
+	}
+
+	ossClient, err := NewAliOSSClient()
+	if err != nil {
+		return nil, err
+	}
+	c.ossClient = ossClient
```
Comment (justinsb) [inline util/pkg/vfs/context.go]: You may want to cache this (if there's some state), it looks like right now we put the value into a field but don't really use it (technically there's a potential threading race)

File: util/pkg/vfs/ossfs.go [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+			resp, err := b.List(opt.prefix, opt.delim, opt.marker, opt.max)
+			if err != nil {
+				if isOSSNotFound(err) {
+					return true, os.ErrNotExist
+				}
+				return false, fmt.Errorf("error listing %s: %v", p, err)
+			}
+
+			conLen := len(resp.Contents)
+			comPrefLen := len(resp.CommonPrefixes)
```
Comment (justinsb) [inline util/pkg/vfs/ossfs.go]: suggestion: I'd probably not declare the variables, and just use `len(resp.Contents)` (for example) directly where we have conLen.  Perf should be unnoticeably similar, and it makes it a little more readable IMO

**Comment by justinsb** [comment]
/lgtm

/ok-to-test

Thanks @xh4n3

**Comment by chrislovecnm** [comment]
We need a rebase, and then we can merge.

File: util/pkg/vfs/context.go (hunk 1 above)
Comment (ihac) [reply util/pkg/vfs/context.go]: Thanks, I've fixed it by reusing `c.ossClient` when it's not nil.

File: Gopkg.lock [hunk 3]
```diff
@@ -424,6 +424,7 @@
     "openstack/identity/v3/tokens",
     "openstack/networking/v2/extensions/security/groups",
     "openstack/networking/v2/extensions/security/rules",
+    "openstack/networking/v2/networks",
```
Comment (ihac) [inline Gopkg.lock]: Seems dep was not `ensure`d in the latest commit?

File: util/pkg/vfs/ossfs.go (hunk 2 above)
Comment (ihac) [reply util/pkg/vfs/ossfs.go]: Agree.

**Pull request merged**